Optimize LayerNormOp

xiaomengy opened the pull request:
Summary:
Optimize LayerNormOp

i-am-not-moving-c2-to-c10

Differential Revision: D14274175

houseroad reviewed (approved):
The failing tests are unrelated. To be safe, we can rebase again to collect more signals.

I would suggest to add the performance improvement in the description of the PR for the record purpose.

xiaomengy committed:
Optimize LayerNormOp (#17604)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/17604

Optimize LayerNormOp

i-am-not-moving-c2-to-c10

Reviewed By: houseroad

Differential Revision: D14274175

fbshipit-source-id: fcc0e908f246d8a47ab220cf006cca2106f731b2
